### ISSUE TITLE
List of all SELECT related issues

### ISSUE THREAD

**Issue by perdjurner** [cfwheels/cfwheels#606]
Let's deal with all SELECT issues at the same time (either in 1.5 or 2.0 if we can't keep it backwards compatible).
- [ ] https://github.com/cfwheels/cfwheels/issues/605
- [ ] https://github.com/cfwheels/cfwheels/issues/601
- [ ] https://github.com/cfwheels/cfwheels/issues/591
- [ ] https://github.com/cfwheels/cfwheels/issues/580
- [ ] https://github.com/cfwheels/cfwheels/issues/328
- [ ] https://github.com/cfwheels/cfwheels/issues/309
- [ ] https://github.com/cfwheels/cfwheels/issues/500 (maybe related)

Will update this issue with a suggestion for how to approach this later.

**Comment by perdjurner** [comment]
Example:

model("author").findAll(include="posts") will return these properties:

id, firstName, lastName, postId, postAuthorId, postTitle

(SQL will be authors.id, authors.firstname, authors.lastname, posts.id AS postId, posts.authorid AS postAuthorId, posts.title AS postTitle)

postAuthorId is a little awkward IMO, we should probably make that just authorId (so it will be the one exception to the rule of always pre-pending the association name name on included models).

**Comment by chrisdpeters** [comment]
@perdjurner I have to disagree with dropping the prefix on the foreign key.

What if you have this scenario?

```
model("author").findAll(include="posts,comments");
=> id, firstName, lastName, postId, postAuthorId, postTitle, commentId, commentAuthorId, commentBody
```

**Comment by perdjurner** [comment]
And the corresponding belongsTo to complete the example:

model("post").findAll(include="author") will return these properties:

id, authorId, title, authorFirstName, authorLastName

(SQL will be posts.id, posts.authorid, posts.title, authors.firstname AS authorFirstName, authors.lastname AS authorLastName)

Note that authorId would in theory be included twice (once for posts.authorId and once due to the rule of pre-pending author to the id in the included author model) but we'll remove all duplicates to fix that.

**Comment by chrisdpeters** [comment]
@perdjurner I just think it would be unexpected. I wouldn't want to see my query dropping any of the columns, even if it seems a little awkward.

**Comment by perdjurner** [comment]
@chrisdpeters Good point, and it doesn't really hurt anything to leave them in there I guess.

**Comment by perdjurner** [comment]
@chrisdpeters should it be postsid, poststitle etc instead? Since "posts" is the association name I mean (author has many posts).

**Comment by chrisdpeters** [comment]
@perdjurner That's a good question. Singularizing it would make it more backwards compatible, right? But man, that would make it more complicated and possibly confusing if you had associations named `posts` and `post`. I hadn't thought about that.

How do you think it should work?

**Comment by perdjurner** [comment]
postTitle makes for more readable code I think.

**Comment by chrisdpeters** [comment]
@perdjurner I think you're right. I hope that doesn't make things too confusing or complicated.

**Comment by perdjurner** [comment]
@neokoenig @chapmandu, your input on this would be appreciated.

**Comment by andybellenie** [comment]
Sorry to wade in a bit late but didn't I do something like this a long time ago? Using the useExpandedColumnAliases setting? I don't recall how it worked but it's in there.

**Comment by andybellenie** [comment]
While we're discussing it I'd really like a setting to choose the column naming convention. I stick to convention whenever possible, but I've had to do a refits recently with database columns using underscores and short of hundreds of property() calls I'm stuck with it. So it would be useful to be able to choose the style of column naming for associations (camelCase, lowerCase, UpperCamelCase, userscores, etc).

**Comment by perdjurner** [comment]
> Sorry to wade in a bit late but didn't I do something like this a long time ago? Using the useExpandedColumnAliases setting? I don't recall how it worked but it's in there.

That's right @andybellenie, we're setting your code to be the default in 2.0.

I think your code prepended the model name though, we're planning on prepending the association name instead so that the developer gets more control over it.

We're also planning to only prepend on included models, I think your code prepended on the current model as well.

**Comment by perdjurner** [comment]
> a setting to choose the column naming convention

That's probably best to discuss in a new issue.

**Comment by chapmandu** [comment]
@perdjurner I prefer the singular prefix `postid` over `postsid`. It feels more in keeping with the readability of associations in general.

**Comment by perdjurner** [comment]
Closing this as we're using the issue tracker for bugs only from now on.
Please implement your feature suggestion as a plugin or make a pull request.
If you need feedback before starting, please consult the Google Group.

Thanks.
